operator/apis/platform/v1alpha1/central_types.go

```diff
@@ -160,6 +165,36 @@ func (c *CentralComponentSpec) GetAdminPasswordGenerationDisabled() bool {
 	return pointer.BoolPtrDerefOr(c.AdminPasswordGenerationDisabled, false)
 }
 
+// Monitoring defines settings for monitoring endpoint.
+type Monitoring struct {
+	// Exposes monitoring endpoint by adding port 9090 with the name
+	// "monitoring" to the deployment, creating service with the same port,
+	// and adding the required network policy.
+	//+operator-sdk:csv:customresourcedefinitions:type=spec,order=1
+	ExposeEndpoint *ExposeEndpoint `json:"exposeEndpoint,omitempty"`
```

Comment:
Looking at @porridge's comment regarding the extensibility of i.e. exposing monitoring for central / scanner  / sensor in the future, would it make sense to do `spec.monitoring.exposeCentralMonitoring` instead, allowing us to later on add other options such as `spec.monitoring.expose(Sensor|Scanner)Monitoring` ? Not sure how we typically deal with this within our spec, whether `spec.monitoring.central.exposeMonitoring` is preferred.

Reply:
I've thought about that. The main reason why I went with this nesting is to keep relevant information in the related scope. If we would go with `spec.monitoring.central.*`, we would have a configuration like this:
```
central: {},
monitoring:
  central: {}
```
with this, we have two sections that describe behaviour of one component. It's not localized.

In the future, if we decide to extend the Monitoring struct with options like port number/name, we can easily do that.

And if we also expose scanner monitoring at some point, we can simply use the same Monitoring struct and benefit from available options. (in theory)